Jenkinsfile

```diff
@@ -0,0 +1,155 @@
+        sh "curl -i -XPOST 'https://greatscott-pinheads-70.c.influxdb.com:8086/write?db=realm' --data-binary 'abi_size,type=${abi} value=${value}i' --user '${env.influx_user}:${env.influx_pass}'"
+    }
+}
+
+def storeJunitResults(String path) {
+    step([
+        $class: 'JUnitResultArchiver',
+        testResults: path
+    ])
+}
+
+def collectAarMetrics() {
+    dir('realm/realm-library/build/outputs/aar') {
+        sh 'unzip realm-android-library-release.aar -d unzipped'
+        sh 'find $ANDROID_HOME -name dx | sort -r | head -n 1 > dx'
+        sh '$(cat dx) --dex --output=temp.dex unzipped/classes.jar'
+        sh 'cat temp.dex | head -c 92 | tail -c 4 | hexdump -e \'1/4 "%d"\' > methods'
+        sendMetrics('methods', readFile('methods'))
+
+        sh 'du -k realm-android-library-release.aar | echo -n `cut -f 1` > aar_size'
+        sendMetrics('aar_size', readFile('aar_size'))
+
+        sh '''for path in unzipped/jni/* ; do
+                abi_size=`du -k $path | cut -f 1`
+                abi=`basename $path`
+                echo -n $abi_size > abi_$abi
+              done
+        '''
+        sendABIMetric('armeabi')
+        sendABIMetric('armeabi-v7a')
+        sendABIMetric('arm64-v8a')
+        sendABIMetric('mips')
+        sendABIMetric('x86')
+        sendABIMetric('x86_64')
+    }
+}
+
+{ ->
+    try {
+        node('FastLinux') {
+            if (isPullRequest()) {
+                checkout([
+                    $class: 'GitSCM',
+                    branches: [[name: "origin/pull/${GITHUB_PR_NUMBER}/head"]],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'CleanCheckout']],
+                    gitTool: 'native git',
+                    submoduleCfg: [],
+                    userRemoteConfigs: [[
+                        credentialsId: '1642fb1a-1a82-4b10-a25e-f9e95f43c93f',
+                        name: 'origin',
+                        refspec: "+refs/heads/master:refs/remotes/origin/master +refs/pull/${GITHUB_PR_NUMBER}/head:refs/remotes/origin/pull/${GITHUB_PR_NUMBER}/head",
+                        url: 'https://github.com/realm/realm-java.git'
+                    ]]
+                ])
+            } else {
+                checkout([
+                    $class: 'GitSCM',
+                    branches: [[name: '*/master']],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'CleanCheckout']],
+                    gitTool: 'native git',
+                    submoduleCfg: [],
+                    userRemoteConfigs: [[url: 'https://github.com/realm/realm-java.git']]
+                ])                
+            }
+
+           stage 'JVM tests'
+           sh 'chmod +x gradlew && ./gradlew assemble check javadoc --stacktrace'
```

Comment:
Might make more sense to just invoke `bash gradlew` rather than chmod'ing all over the place.

```
def gradle(command) {
  sh "bash ./gradlew ${command}"
}
```